File: java-testapp/src/main/java/com/fsryan/tools/logging/testappjava/SysOutEventLogger.java
package com.fsryan.tools.logging.testappjava;

import com.fsryan.tools.logging.FSEventLogger;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;

import java.util.Map;

public class SysOutEventLogger implements FSEventLogger {

    @Override
    public String id() {
        return "sysout";
    }

    @Override
    public void addAttr(@Nonnull String attrName, @Nonnull String attrValue) {
        System.out.println("add attr " + attrName + "=" + attrValue);
    }

    @Override
    public void removeAttr(@Nonnull String attrName) {
        System.out.println("remove attr " + attrName);
    }

    @Override
    public void incrementAttrValue(@Nonnull String attrName) {
        System.out.println("incrementCountableAttr " + attrName);
    }

    @Override
    public void addEvent(@Nonnull String eventName, @Nonnull Map<String, String> attrs) {
        System.out.println("addEvent " + eventName + "; attrs = " + attrs);
    }

    @Override
    public boolean runInTestEnvironment() {
        return false;
    }

    @Override
    public void sendTimedOperation(
            @Nonnull String operationName,
            long startTimeMillis,
            long endTimeMillis,
            @Nullable String durationAttrName,
            @Nullable String startTimeMillisAttrName,
            @Nullable String endTimeMillisAttrName,
            @Nonnull Map<String, String> startAttrs,
            @Nonnull Map<String, String> endAttrs
    ) {
        System.out.println("timed operation " + operationName + "; startMillis = " + startTimeMillis + "; endMillis = " + endTimeMillis + "; startAttrs = " + startAttrs + "; endAttrs = " + endAttrs);
    }
}

File: kotlin-testapp/src/main/java/com/fsryan/tools/logging/testappkt/SysOutEventLogger.kt
package com.fsryan.tools.logging.testappkt

import com.fsryan.tools.logging.FSEventLogger

class SysOutEventLogger : FSEventLogger {
    override fun id(): String = "sysout"
    override fun addAttr(attrName: String, attrValue: String) = println("add attr $attrName=$attrValue")
    override fun removeAttr(attrName: String) = println("remove attr $attrName")
    override fun incrementAttrValue(attrName: String) = println("incrementCountableAttr $attrName")
    override fun addEvent(eventName: String, attrs: Map<String, String>) = println("addEvent $eventName; attrs = $attrs")
    override fun sendTimedOperation(
        operationName: String,
        startTimeMillis: Long,
        endTimeMillis: Long,
        startAttrs: Map<String, String>,
        endAttrs: Map<String, String>
    ) {
        println("timed operation $operationName; startMillis = $startTimeMillis; endMillis = $endTimeMillis; startAttrs = $startAttrs; endAttrs = $endAttrs")
    }
}